src/legacy/ui/public/agg_types/buckets/date_histogram.js

```diff
@@ -118,18 +119,22 @@ export const dateHistogramBucketAgg = new BucketAggType({
     },
     {
       name: 'interval',
-      type: 'optioned',
-      deserialize: function (state) {
+      editorComponent: TimeIntervalParamEditor,
+      deserialize: function (state, agg) {
+        // For upgrading from 7.0.x to 7.1.x - intervals are now stored as key of options or custom value
+        if (state === 'custom') return _.get(agg, 'params.customInterval');
```

Comment:
I think our coding guidelines discourage the use of single-line if statements. This should probably be:

```js
if (state === 'custom') {
  return _.get(agg, 'params.customInterval');
}
```